mistralrs-core/src/pipeline/gguf_tokenizer.rs

```diff
@@ -113,6 +113,34 @@ pub fn convert_ggml_to_hf_tokenizer(content: &Content) -> Result<ConversionResul
 
             (tokenizer, "unigram")
         }
+        "gpt2" => {
+            // This is a `bpe` tokenizer
+            let merges = merges
+                .as_ref()
+                .expect("Expect `tokenizer.ggml.merges` for `llama` unigram tokeizer.")
+                .into_iter()
+                .map(|merges| {
+                    let res = merges.splitn(2, ' ').collect::<Vec<_>>();
+                    (res[0].to_string(), res[1].to_string())
+                })
+                .collect::<Vec<_>>();
+            let mut vocab = HashMap::new();
+            for (i, token) in tokens.iter().enumerate() {
+                vocab.insert(token.clone(), i as u32);
+            }
+
+            let bpe = BpeBuilder::new()
+                .vocab_and_merges(vocab, merges)
+                .build()
+                .map_err(anyhow::Error::msg)?;
+            let mut tokenizer = Tokenizer::new(ModelWrapper::BPE(bpe));
+            tokenizer.with_decoder(decoders::byte_level::ByteLevel::new(true, true, true));
+
+            tokenizer.add_special_tokens(&[AddedToken::from(tokens[bos as usize].clone(), true)]);
+            tokenizer.add_special_tokens(&[AddedToken::from(tokens[eos as usize].clone(), true)]);
```

Comment:
Just curious, BPE has support for [setting `unk` in it's builder variant](https://docs.rs/tokenizers/latest/tokenizers/models/bpe/struct.BpeBuilder.html#method.unk_token), is it not relevant for some reason? (_I know very little about these things_)

Reply:
@polarathene the GGUF file I am testing with (QuantFactory/Meta-Llama-3-8B-Instruct-GGUF) does not have a unk metadata, so I left it out here.

Reply:
Sure, but what about when they do? I assume that's possible since the tokenizer builder for BPE does support setting `unk`? There is no check for this, so if there was it'd just ignore it and introduce a bug?

Reply:
Yes, I wasn't sure if this is guaranteed to be not provided, but just in case I added [8d4dba5](https://github.com/EricLBuehler/mistral.rs/pull/397/commits/8d4dba52462c9e8002767c654946e3f043a2872e) and [763241e](https://github.com/EricLBuehler/mistral.rs/pull/397/commits/763241e16ad10379eff8a4e9f03bcf48ad2fd85e).